File: app/src/main/java/me/gilo/wc/repo/CategoryRepository.java
package me.gilo.wc.repo;


import me.gilo.wc.common.WooLiveData;
import me.gilo.woodroid.Woocommerce;
import me.gilo.woodroid.models.Category;
import me.gilo.woodroid.models.filters.ProductCategoryFilter;

import javax.inject.Inject;
import java.util.List;

public class CategoryRepository {

    @Inject
    Woocommerce woocommerce;

    @Inject
    public CategoryRepository() {
    }


    public WooLiveData<Category> create(Category category) {
        final WooLiveData<Category> callBack = new WooLiveData();

        woocommerce.CategoryRepository().create(category).enqueue(callBack);
        return callBack;
    }


    public WooLiveData<Category> category(int id) {
        final WooLiveData<Category> callBack = new WooLiveData();

        woocommerce.CategoryRepository().category(id).enqueue(callBack);
        return callBack;
    }

    public WooLiveData<List<Category>> categories() {
        final WooLiveData<List<Category>> callBack = new WooLiveData();

        woocommerce.CategoryRepository().categories().enqueue(callBack);
        return callBack;
    }

    public WooLiveData<List<Category>> categories(ProductCategoryFilter productCategoryFilter) {
        final WooLiveData<List<Category>> callBack = new WooLiveData();

        woocommerce.CategoryRepository().categories(productCategoryFilter).enqueue(callBack);
        return callBack;
    }

    public WooLiveData<Category> update(int id, Category category) {
        final WooLiveData<Category> callBack = new WooLiveData();

        woocommerce.CategoryRepository().update(id, category).enqueue(callBack);
        return callBack;
    }

    public WooLiveData<Category> delete(int id) {
        final WooLiveData<Category> callBack = new WooLiveData();

        woocommerce.CategoryRepository().delete(id).enqueue(callBack);
        return callBack;
    }

    public WooLiveData<Category> delete(int id, boolean force) {
        final WooLiveData<Category> callBack = new WooLiveData();

        woocommerce.CategoryRepository().delete(id, force).enqueue(callBack);
        return callBack;
    }


}

File: app/src/main/java/me/gilo/woodroid/app/repo/CategoryRepository.kt
package me.gilo.woodroid.app.repo


import me.gilo.woodroid.app.common.WooLiveData
import me.gilo.woodroid.Woocommerce
import me.gilo.woodroid.models.Category
import me.gilo.woodroid.models.filters.ProductCategoryFilter

import javax.inject.Inject

class CategoryRepository @Inject
constructor() {

    @Inject
    lateinit var woocommerce: Woocommerce


    fun create(category: Category): WooLiveData<Category> {
        val callBack = WooLiveData<Category>()

        woocommerce.CategoryRepository().create(category).enqueue(callBack)
        return callBack
    }


    fun category(id: Int): WooLiveData<Category> {
        val callBack = WooLiveData<Category>()

        woocommerce.CategoryRepository().category(id).enqueue(callBack)
        return callBack
    }

    fun categories(): WooLiveData<List<Category>> {
        val callBack = WooLiveData<List<Category>>()

        woocommerce.CategoryRepository().categories().enqueue(callBack)
        return callBack
    }

    fun categories(productCategoryFilter: ProductCategoryFilter): WooLiveData<List<Category>> {
        val callBack = WooLiveData<List<Category>>()

        woocommerce.CategoryRepository().categories(productCategoryFilter).enqueue(callBack)
        return callBack
    }

    fun update(id: Int, category: Category): WooLiveData<Category> {
        val callBack = WooLiveData<Category>()

        woocommerce.CategoryRepository().update(id, category).enqueue(callBack)
        return callBack
    }

    fun delete(id: Int): WooLiveData<Category> {
        val callBack = WooLiveData<Category>()

        woocommerce.CategoryRepository().delete(id).enqueue(callBack)
        return callBack
    }

    fun delete(id: Int, force: Boolean): WooLiveData<Category> {
        val callBack = WooLiveData<Category>()

        woocommerce.CategoryRepository().delete(id, force).enqueue(callBack)
        return callBack
    }


}
